SingularityUI/app/views/request.coffee

```diff
@@ -61,11 +61,18 @@ class RequestView extends View
             app.router.navigate 'requests', trigger: true
 
     runRequest: (e) =>
-        @model.promptRun =>
-            @trigger 'refreshrequest'
-            setTimeout =>
+        @model.promptRun (data) =>   
+            # If user wants to redirect to a file after the task starts
+            if data.filename.length > 1
+                @listenTo @activeTasks, 'add', @redirectToFile
```

Comment:
Yep we can do `listenToOnce`, but I also already call stopListening in the `stopPollingTask` method (line 116) which stops the event listener in the case that they cancel the redirect (as well as if they redirect). So we can change that, but I'll still be calling the `stopPollingTask` to cancel the interval, so it won't really have any noticeable effect, if I'm thinking about this right.